aws/aws.go

```diff
@@ -19,6 +19,15 @@ type Credentials struct {
 	Expiration      time.Time
 }
 
+type Profile struct {
+	Name         string
+	LifetimeLeft time.Duration
+}
+
+type Profiles struct {
```

Comment:
Looks like the only field in `Profiles` is a slice of `Profile`s. In this case there is no need to create another custom type, we can simply create a slice of `Profile`s instead of initializing a `Profiles` struct:

```go
profiles := []Profile{}

// Add stuff to slice.
for _, x := range whatever {
    if stuff {
        profiles := append(profiles, x)
    }
}
```

The `Profiles` type only adds unnecessary wrapping in this case.